src/js/client/common-dialogs-widget.js

```diff
@@ -0,0 +1,125 @@
+fluid.registerNamespace("gpii.devpmt.dialogs");
+
+/**
+ * DevPMT Base Dialog
+ *
+ * Creates the simplest dialog, currently based on the foundation
+ * modal dialog. Can be inherited to create more complex dialogs.
+ */
+fluid.defaults("gpii.devpmt.dialogs.baseDialog", {
+    gradeNames: ["gpii.handlebars.templateAware"],
+    templates: {
+        initial: "common-dialog-widget"
+    },
+    model: {},
+    invokers: {
+        renderInitialMarkup: {
+            func: "{that}.renderMarkup",
+            args: ["initial", "{that}.options.templates.initial", "{that}.model"]
+        },
+        openDialog: {
+            funcName: "gpii.devpmt.dialogs.baseDialog.openDialog",
+            args: ["{that}", "{that}.dom.dialogContainer"]
+        },
+        closeDialog: {
+            funcName: "gpii.devpmt.dialogs.baseDialog.closeDialog",
+            args: ["{that}", "{that}.dom.dialogContainer"]
+        }
+    },
+    selectors: {
+        dialogContainer: "#dialog-container"
+    },
+    listeners: {
+        "onMarkupRendered.openDialog": {
+            func: "{that}.openDialog"
+        }
+    },
+    events: {
+        // Sometimes foundation can move around the markup for the
+        // dialog to float it, etc. and we can loose event bindings.
+        // This allows activity to happen immediately after the dialog
+        // is opened.
+        afterOpenDialog: null
+    }
+});
+
+gpii.devpmt.dialogs.baseDialog.openDialog = function (that, dialogContainer) {
+    var elem = new Foundation.Reveal(dialogContainer, {
+        appendTo: that.options.selectors.initial
+    });
+    dialogContainer.on("closed.zf.reveal", function () {
+        that.destroy();
+    });
+    dialogContainer.foundation("open");
+    that.events.afterOpenDialog.fire();
+};
+
+gpii.devpmt.dialogs.baseDialog.closeDialog = function (that, dialogContainer) {
+    dialogContainer.foundation("close");
+};
+
+/**
+ * DevPMT Confirmation Dialog
+ *
+ * Suitable dialog for performing actions that require either an
+ * 'accept' or 'cancel' type answer. Such as adding a product to
+ * an NP Set, deleting a context from an NP Set, etc.
+ *
+ * To create confirmation dialogs, inherit this grade. Then the
+ * following items need to be overriden:
+ *
+ * - Create a handlebars template in the standard partials area
+ * - Override the `templates.initial` option with that template.
+ * - The template need only retain the class names for the open
+ *   and close buttons in this grades selectors, to allow wiring.
+ * - You can optionally override the `acceptConfirmDialog` and
+ *   `cancelConfirmDialog` invokers with custom behavior, and
+ *   then close the dialog if your processing is successful.
+ */
+fluid.defaults("gpii.devpmt.dialogs.confirmDialog", {
+    gradeNames: ["gpii.devpmt.dialogs.baseDialog"],
+    templates: {
+        initial: "common-confirm-dialog-widget"
+    },
+    selectors: {
+        acceptButton: ".confirm-dialog-button",
```

Comment:
Guidelines on class names: https://docs.fluidproject.org/infusion/development/ClassNameConventions.html
Needn't particularly use fl or flc but there should be some kind of project-specific prefix

Reply:
I've prefixed all my selectors with pmt, but kept them a bit shorter than the core infusion standard.  :)